core/domain/customization_args_util.py

```diff
@@ -91,33 +90,35 @@ def validate_customization_args_and_values(
             'Expected customization args to be a dict, received %s'
             % customization_args)
 
-    # Validate and clean up the customization args.
-
-    # Populate missing keys with the default values.
-    customization_args = get_full_customization_args(
-        customization_args, ca_specs_to_validate_against)
-
-    # Remove extra keys.
-    extra_args = []
+    # Check for extra invalid keys.
     for arg_name in customization_args.keys():
         if not isinstance(arg_name, python_utils.BASESTRING):
             raise utils.ValidationError(
                 'Invalid customization arg name: %s' % arg_name)
         if arg_name not in ca_spec_names:
-            extra_args.append(arg_name)
-            logging.warning(
+            raise utils.ValidationError(
                 '%s %s does not support customization arg %s.'
                 % (item_name.capitalize(), item_type, arg_name))
-    for extra_arg in extra_args:
-        del customization_args[extra_arg]
 
     # Check that each value has the correct type.
     for ca_spec in ca_specs_to_validate_against:
+        if ca_spec.name not in customization_args:
+            raise utils.ValidationError(
+                'Customization argument is missing key: %s' % ca_spec.name)
         try:
+            # Since SubtitledHtml is a custom object which contains a html
+            # field as one of the properties in its dict schema, we use
+            # apply_custom_validators=False so that normalize_against_schema
+            # traverses to the nested html schema inside SubtitledHtml and
+            # properly normalizes the html.
+            # Additionally, default values might not pass custom validation
```

Comment:
>> So why not implement it, instead?

> I originally didn't want to repeat code, thought it was unnecessary to have the html validation repeated in SubtitledHtml in objects.py.
>
> I found another instance of using apply_custom_validators=False, I think it is also relevant here. The comment from the other instance is:
>
```
            # The default value might not pass validation checks (e.g. the
            # Image component has a required field whose default value is
            # empty). Thus, when checking the default value schema, we don't
            # apply the custom validators.
```
> This also applies in this situation, since our migration could potentially populate missing customization arguments 
> with their default values. Adding the above comment to my comment as well.

I still don't understand the first part. Validation and normalization are different things, conceptually. Why would the additional validation stop the inner normalization from happening?